File: palindrome/Palindrome.java
/** Contains a given String a palindrome ? */
class Palindrome {

  /* returns true if candidate is palindrome, otherwise false */
  private boolean isPalindrome(String candidate) {
    if (candidate.length() < 1) { return false; }
    for (int left=0, right=candidate.length()-1; left < right; right--, left++) {
      if (candidate.charAt(left) != candidate.charAt(right)) {
        return false;
      }
    }
    return true;
  }

  /* writes to console if candidate is palindrome or not */
  void testForPalindrome(String candidate) {
    if (isPalindrome(candidate)) System.out.println(candidate + " is palindrome");
    else System.out.println(candidate + " is NOT palindrome");
  }

  /* entry */
  public static void main(String[] args) {
    Palindrome checker = new Palindrome();
    checker.testForPalindrome("");
    checker.testForPalindrome("abc");
    checker.testForPalindrome("a");
    checker.testForPalindrome("aba");
    checker.testForPalindrome("abba");
  }
}

File: palindrome/palindrome.cpp
/** Contains a given String a palindrome ? */

#include <string>
#include <iostream>

using namespace std;

/* returns true if candidate is palindrome, otherwise false */
bool isPalindrome(string candidate) {
  if (candidate.length() < 1) return false;

  int left = 0;
  int right = candidate.length()-1;

  while (left <= right) {
    if (candidate[left] != candidate[right]) return false;
    left++;
    right--;
  }

  return true;
}

/* writes to stdout if candidate is palindrome or not */
void testForPalindrome(string candidate) {
  if (isPalindrome(candidate)) {
    cout << candidate << " is palindrome" << endl;
  } else {
    cout << candidate << " is NOT palindrome" << endl;
  }
}

/* entry */
int main() {
  testForPalindrome("");
  testForPalindrome("a");
  testForPalindrome("abc");
  testForPalindrome("aba");
  testForPalindrome("abba");
  return 0;
}